core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/EagerZippedStream.scala

```diff
@@ -0,0 +1,24 @@
+package com.lynbrookrobotics.potassium.streams
+
+class EagerZippedStream[A, B] extends Stream[(A, B)] {
+  override val expectedPeriodicity: ExpectedPeriodicity = NonPeriodic
+
+  private[this] var lastASlot : Option[A] = None
+  private[this] var lastBSlot : Option[B] = None
+
+  def maybePublish(): Unit = {
+    if (lastASlot.isDefined && lastBSlot.isDefined) {
+      publishValue((lastASlot.get, lastBSlot.get))
+    }
+  }
+
+  def receiveA(value: A): Unit = {
+    lastASlot = Some(value)
+    maybePublish()
+  }
+
+  def receiveB(value: B): Unit = {
+    lastBSlot = Some(value)
+    maybePublish()
+  }
+}
```

Comment:
Does not have strong reference to parent streams?
data.zip(otherData.map(...))
will break stuff cause there is are no strong references to otherData.map(...)